client/app/components/group/_groupMap/groupMap.controller.js

```diff
@@ -0,0 +1,127 @@
+class GroupMapController {
+  constructor($scope, CurrentGroup, CurrentStores, CurrentUsers, $timeout) {
+    "ngInject";
+    Object.assign(this, {
+      $scope,
+      CurrentGroup,
+      CurrentStores,
+      CurrentUsers,
+      $timeout,
+      markers: {},
+      bounds: {},
+      center: {},
+      defaults: {
+        scrollWheelZoom: false
+      },
+      options: {
+        showUsers: false
+      }
+    });
+  }
+
+  $onInit() {
+    // deep watching
+    this.destroy = this.$scope.$watch(() => this.CurrentStores.list, () => {
+      this.$timeout(() => this.update(), 200);
```

Comment:
It could have to do with the watch settings in ui-leaflet. Maybe play around with the `watchOptions` setting? Although `markers` always gets replaced with a new object, so even the most shallow watching should work.
https://github.com/angular-ui/ui-leaflet/blob/master/CHANGELOG.md#breaking-changes